src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -110,44 +121,126 @@ protected override void Dispose(bool disposing)
 
         #region Private methods
 
-        private void StartRequest(RequestCompletionSource state)
+        private async void StartRequest(RequestCompletionSource state)
```

Comment:
It's difficult to know whether the TCS is needed at all, as this is library is being built up in pieces and I don't have a good sense of where it's ending, but as it's currently written it still seems like a TCS is needed somewhere, due to the underlying async APIs in libcurl being used... those will need to signal something and cause some Task to complete.

That said, it's unclear to me why we're both calling StartNew and using an async method... why can't we just use the latter and invoke it directly from SendAsync?  Since everything is getting marshaled to the TCS, with a catch block used to catch everything, and since the Task returned from StartNew is getting ignored, it probably does make sense for this to be an async void method in that case, since no one's going to be observing the resulting Task anyway.